pkg/client/client.go

```diff
@@ -1723,25 +1723,8 @@ func (c *Client) CreateOrUpdateServiceMonitor(ctx context.Context, sm *monv1.Ser
 }
 
 func (c *Client) CreateOrUpdateAPIService(ctx context.Context, apiService *apiregistrationv1.APIService) error {
-	apsc := c.aggclient.ApiregistrationV1().APIServices()
-	existing, err := apsc.Get(ctx, apiService.GetName(), metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		_, err = apsc.Create(ctx, apiService, metav1.CreateOptions{})
-		return errors.Wrap(err, "creating APIService object failed")
-	}
-	if err != nil {
-		return errors.Wrap(err, "retrieving APIService object failed")
-	}
-
-	required := apiService.DeepCopy()
-	required.ResourceVersion = existing.ResourceVersion
-	if val, ok := required.Annotations["service.beta.openshift.io/inject-cabundle"]; ok && val == "true" {
-		if len(existing.Spec.CABundle) > 0 {
-			required.Spec.CABundle = existing.Spec.CABundle
-		}
-	}
-	_, err = apsc.Update(ctx, required, metav1.UpdateOptions{})
-	return errors.Wrap(err, "updating APIService object failed")
+	_, _, err := resourceapply.ApplyAPIService(ctx, c.aggclient.ApiregistrationV1(), c.eventRecorder, apiService)
```

Comment:
deleting a label/annotation, would require two steps https://github.com/openshift/library-go/blob/5674ec6ced1c6e6ddc86201e5232d6713c7a262c/pkg/operator/resource/resourcemerge/object_merger.go#L141

+

labels + annotations will be merged now, as we do in `CreateOrUpdateServiceAccount` e.g.